cf-app/src/main/kotlin/space/crowdforce/service/tg/ArgumentValidator.kt

```diff
@@ -0,0 +1,102 @@
+package space.crowdforce.service.tg
+
+import org.springframework.stereotype.Component
+import space.crowdforce.domain.Project
+import space.crowdforce.domain.User
+import space.crowdforce.repository.ProjectRepository
+import space.crowdforce.service.tg.Argument.*
+import space.crowdforce.service.tg.Navigation.toInnerLink
+import space.crowdforce.service.tg.command.CommandAnswer
+import space.crowdforce.service.tg.command.Link
+
+interface ArgumentValidator {
+    fun validate(user: User, context: UserContext): CommandAnswer?
+
+    fun argument(): Argument
+}
+
+@Component
+class OwnedProjectId(val projectRepository: ProjectRepository) : ArgumentValidator {
+    override fun validate(user: User, context: UserContext): CommandAnswer? {
+        val value = context.value(argument())
+                ?: return validationFail(user)
+
+        val project = projectRepository.findById(value.toInt())
+
+        if (project == null || project.ownerId != user.id)
+            return validationFail(user)
```

Comment:
minor: I'd tell to the user that the project not found (or the user doesn't have permissions / isn't owner)